src/codelabs/iterables.md

```diff
@@ -0,0 +1,608 @@
+If more than one or no element in the `Iterable` satisfies the predicate condition, then the
+method throws a [StateError][StateError class] exception.
+
+{{site.alert.warn}}
+  `singleWhere()` steps through the whole `Iterable` until the last element, which can cause
+  problems if the `Iterable` is infinite or contains a large collection of elements.
+{{site.alert.end}}
+
+Your goal is to implement the predicate for `singleWhere()` that satisfies the following conditions:
+
+* The element contains the character `'a'`.
+* The element starts with the character `'M'`.
+
+All the elements in the test data are [strings;][String class]
+you can check the class documentation for help.
+
+[//]: https://gist.github.com/241c6bc591f9436a9be0116724222953
+<iframe
+  src="{{site.dartpad-embed}}?id=241c6bc591f9436a9be0116724222953&theme=dark"
+  frameborder="no"
+  height="250"
+  width="100%">
+</iframe>
+
+## Checking conditions
+
+When working with `Iterable`, sometimes you need to verify that the elements of a collection
+satisfy some condition.
+
+You might be tempted to write a solution using a `for-in` loop like this one:
+
+{:.bad}
+<?code-excerpt "iterables/test/iterables_test.dart (every-bad)"?>
+{% prettify dart %}
+for (var item in items) {
+  if (item.length < 5) {
+    return false;
+  }
+}
+return true;
+{% endprettify %}
+
+However, you can accomplish the same using the `every()` method:
+
+<?code-excerpt "iterables/test/iterables_test.dart (every-good)"?>
+{% prettify dart %}
+return items.every((element) => element.length >= 5);
+{% endprettify %}
+
+Using the `every()` method results in code that is more readable, compact, and less error prone.
+
+### Example: Using any() and every()
+
+The `Iterable` class provides two methods that you can use to verify conditions:
+
+* `any()`: Returns true if at least one element satisfies the condition.
+* `every()`: Returns true if all elements satisfy the condition.
+
+Run this exercise to see them in action.
+
+[//]: https://gist.github.com/d56963729339cea951e16209e0a26e4c
+<iframe
+  src="{{site.dartpad-embed}}?id=d56963729339cea951e16209e0a26e4c"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px"
+  frameborder="no"
+  height="375"
+  width="100%" >
+</iframe>
+
+In the example, `any()` verifies that at least one element contains the
+character `a`, and `every()` verifies that all elements have a length equal to 
+or greater than 5.
+
+After running the code, try changing the predicate of `any()` so it
+returns false:
+
+<?code-excerpt "iterables/test/iterables_test.dart (any-false)"?>
+{% prettify dart %}
+if (items.any((element) => element.contains('Z'))) {
+  print('At least one element contains "Z"');
+} else {
+  print('No element contains "Z"');
+}
+{% endprettify %}
+
+You can also use `any()` to verify that no element of an `Iterable`
+matches a certain condition.
+
+
+### Exercise: Verify that an Iterable satisfies a condition
```

Comment:
Nit: I noticed that the solution code for this exercise is missing :
```
class User {
  String name;
  int age;

  User(
    this.name,
    this.age,
  );
}
```

It's probably best if all solution code makes the test pass on its own (i.e. when users copy + paste solution code into the "Dart" editor, they should be able to run the Exercise and see the tests pass)

Reply:
I've fixed the gist.